### PR TITLE
CC | kata-deploy-push more space

### PR DESCRIPTION
**Pull request by jepio**
Remove some tools from runner and some temp files during build so that we (hopefully) don't run out of disk space so often.

### CODE REVIEW

**Comment by stevenhorsman** [comment]
The non-TEE k8s test failed on:
```
not ok 12 [cc][agent][kubernetes][containerd] Test cannot pull an image from an authenticated registry without credentials
```
but the logs show it didn't ever get to doing the pull, so I've requeued to see what happens next time.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

**Comment by stevenhorsman** [comment]
Given that only the tdx test is failing, which has a known issue, once we get a second review I think we can merge this.

**Review by Tim-0731-Hzt — APPROVED** [review]
Thanks @jepio LGTM

**Pull request merged**
